W-12559690-externaldns-k8-duke (#229)
mulesoft/docs-gateway: gateway/1.4/modules/ROOT/pages/flex-gateway-k8-external-dns.adoc

Reviewer: Looks good to me!

Author: Thx, I've committed everything. Can you approve the PR, @ahejeij ?

= Provide a Domain Name for Your Kubernetes Service

> Use ExternalDNS to expose your Kubernetes service through a domain name instead of a difficult-to-understand external IP or random host name. 

Author: @ahejeij  Is "Kubernetes service" correct? What do we mean exactly?

I'd like to make it more clear in the intro how this relates specifically to Flex. 

Are we talking about a service for the pod (or pods) where Flex runs? Meaning the service that allows that pod to be assigned a name and unique IP address (clusterIP). And we're showing how to set a domain name for that that IP? Or what? Thx!

Reviewer: It refers to your kubernetes application let's say. You have an app (or service) that you want to expose through flex gateway

> = Provide a Domain Name for Your Kubernetes Service

Author: TODO: title change?
```suggestion
= Provide a Domain Name for an Application on Your Kubernetes Cluster
```

Author: @ahejeij I'll commit this if you are ok with it.

Reviewer: go ahead!

> Use ExternalDNS to expose your Kubernetes service through a domain name instead of a difficult-to-understand external IP or random host name. 

Author: TODO: "Kubernetes service"?
```suggestion
Use ExternalDNS to expose your Kubernetes service through a domain name instead of a difficult-to-understand external IP or random host name. 
```

> Use ExternalDNS to expose your Kubernetes service through a domain name instead of a difficult-to-understand external IP or random host name. 

Author: ```suggestion
Use ExternalDNS to expose your an application or service on your Kubernetes cluster through a domain name instead of a difficult-to-understand external IP or random host name. 
```

Author: @ahejeij And I'll commit this if you're ok with it. Thx!

> Use ExternalDNS to expose your an application or service on your Kubernetes cluster through a domain name instead of a difficult-to-understand external IP or random host name. 

Author: ```suggestion
Use ExternalDNS to expose an application or service on your Kubernetes cluster through a domain name instead of a difficult-to-understand external IP or random host name. 
```

> https://github.com/kubernetes-sigs/external-dns/blob/master/README.md[ExternalDNS^] is an open source project that supports DNS configurations on many DNS providers. The tool is _not_ a DNS server or provider. Instead, ExternalDNS discovers Kubernetes resources through public DNS servers and configures the resources for DNS providers.

Author: ```suggestion
https://github.com/kubernetes-sigs/external-dns/blob/master/README.md[ExternalDNS^] is an open source project that supports DNS configurations on many DNS providers. The tool is _not_ a DNS server or provider. Instead, ExternalDNS discovers Kubernetes resources through public DNS servers and configures the resources for supported DNS providers.
```

> This guide uses ExternalDNS to configure domain names in an AWS Route 53 provider, which one of the stable providers supported by ExternalDNS. 

Reviewer 2: which is one

Author: done

> This guide uses ExternalDNS to configure domain names in an AWS Route 53 provider, which one of the stable providers supported by ExternalDNS. 

Author: ```suggestion
This guide uses ExternalDNS to configure domain names in an AWS Route 53 provider, which is one of the stable providers supported by ExternalDNS. 
```

> The steps in this guide use ExternalDNS to configure domain names in an AWS Route 53 provider, which one of the stable providers supported by ExternalDNS. 

Author: ```suggestion
This guide uses ExternalDNS to configure domain names in an AWS Route 53 provider, which one of the stable providers supported by ExternalDNS. 
```

+
The option is critical for making `external-dns` work as intended.

** `- --policy=upsert-only`: Prevent ExternalDNS from deleting any records. Omit this option to enable full synchronization.

** `- --aws-zone-type=public`: Make only public hosted zones visible. Valid values are `public`, `private`, or no value for both types of zones.

* `env` option `value: <aws-region>`: Set the region in which your EKS cluster is installed, for example, `us-east-1`.

. If you are using static credentials, remove the comments (`#`) from the lines that contain hidden `env` and `volumes` configurations.

. Apply the configuration to your cluster by using the following command from a terminal window:
+
[source,kubctl]
----
kubectl create --filename <external-dns-yaml> --namespace <your-namespace>
----

== Set DNS Names through Your Helm Chart for Flex Gateway

Use the `extraAnnotations` parameter with `helm upgrade` to set the domain names. Provide an annotation that specifies an ExternalDNS key with your gateway domains:

----
--set extraAnnotations."external-dns\.alpha\.kubernetes\.io/hostname"=<domain-1>,<domain-2>...
----

> The `helm upgrade` command options to use depends on whether your registered gateway is deployed to a cluster:

Author: ```suggestion
The `helm upgrade` command options to use depend on whether your registered gateway is deployed to a cluster:
```

> 

Reviewer 2: options ... depend on whether

Author: oops, thx! 😆

Author: done
